tests/system/src/test/java/test/javafx/embed/swing/SwingNodeDnDMemoryLeakTest.java

```diff
@@ -65,12 +66,14 @@ public static void teardownOnce() {
     }
 
     @Test
-    public void testSwingNodeMemoryLeak() {
+    public void testSwingNodeMemoryLeak() throws InvocationTargetException, InterruptedException {
         Util.runAndWait(() -> {
             testSwingNodeObjectsInStage();
         });
-        attemptGCSwingNode();
-        assertEquals(TOTAL_SWINGNODE, getCleanedUpSwingNodeCount());
+
+        for (WeakReference<SwingNode> ref : weakRefArrSN) {
```

Comment:
I wonder if we should add a utility method to JMemoryBuddy to operate on an array/collection, so we don't have to repeat this code again and again?

Reply:
Done